pkg/devfile/parser/reader.go

```diff
@@ -38,6 +39,8 @@ type YamlSrc struct {
 	Path string
 	// URL of the yaml file
 	URL string
+	// Token to access a private URL like a private repository
+	Token string
```

Comment:
before I start reviewing, I'm wondering if by-design we can ONLY accept a single token instead of a token list? 
saw you have the `readkubernetesyaml` with the token passed in. led me to think of the case that the private repo A contains devfile.yaml, which has a kubernetes component reference to a private repo B. In this case, only one token can be passed in, it's gonna cause problem.

Reply:
I think that is a valid question but which is currently out of scope for this issue. We have the same issue in HAS with SPI where we only support one token and if someone's devfile has multiple private repo then its something we dont support now.